src/SharpLearning.Backend.Cntk.Test/CntkRawMnistTest.cs

```diff
@@ -0,0 +1,575 @@
+            int strideW = 2, int strideH = 2, string outputName = "")
+        {
+            if (input.Shape.Rank != 3)
+            {
+                throw new ArgumentException("Pool2D layer requires shape rank 3, got rank " + input.Shape.Rank);
+            }
+
+            return CNTKLib.Pooling(input, PoolingType.Max,
+                new int[] { poolW, poolH }, new int[] { strideW, strideH });
+        }
+
+        public static Function Dropout(Variable input, double dropoutRate, uint seed = 465)
+        {
+            return CNTKLib.Dropout(input, dropoutRate, seed);
+        }
+
+        public static Function BatchNormalizationLayer(Variable input, bool spatial,
+            double initialScaleValue = 1, double initialBiasValue = 0, int bnTimeConst = 5000)
+        {
+            var biasParams = new Parameter(new int[] { NDShape.InferredDimension }, (float)initialBiasValue, Device, "");
+            var scaleParams = new Parameter(new int[] { NDShape.InferredDimension }, (float)initialScaleValue, Device, "");
+            var runningMean = new Constant(new int[] { NDShape.InferredDimension }, 0.0f, Device);
+            var runningInvStd = new Constant(new int[] { NDShape.InferredDimension }, 0.0f, Device);
+            var runningCount = Constant.Scalar(0.0f, Device);
+
+            return CNTKLib.BatchNormalization(input, scaleParams, biasParams, runningMean, runningInvStd, runningCount,
+                spatial, (double)bnTimeConst, 0.0, 1e-5 /* epsilon */);
+        }
+    }
+
+    // Below copied from: https://github.com/migueldeicaza/TensorFlowSharp/blob/master/Learn/Datasets/MNIST.cs
+    // Do NOT use this code as foundation for other things, quality is not great!
+
+    // Stores the per-image MNIST information we loaded from disk 
+    //
+    // We store the data in two formats, byte array (as it came in from disk), and float array
+    // where each 0..255 value has been mapped to 0.0f..1.0f
+    public struct MnistImage
+    {
+        public int Cols, Rows;
+        public byte[] Data;
+        public float[] DataFloat;
+
+        public MnistImage(int cols, int rows, byte[] data)
+        {
+            Cols = cols;
+            Rows = rows;
+            Data = data;
+            DataFloat = new float[data.Length];
+            for (int i = 0; i < data.Length; i++)
+            {
+                DataFloat[i] = Data[i] / 255f;
+            }
+        }
+    }
+
+    // Helper class used to load and work with the Mnist data set
+    public class Mnist
+    {
+        // 
+        // The loaded results
+        //
+        public MnistImage[] TrainImages, TestImages, ValidationImages;
+        public byte[] TrainLabels, TestLabels, ValidationLabels;
+        public byte[,] OneHotTrainLabels, OneHotTestLabels, OneHotValidationLabels;
+
+        public BatchReader GetTrainReader(int readFromItem) => new BatchReader(TrainImages, TrainLabels, OneHotTrainLabels, readFromItem);
```

Comment:
Added the possibility of creating a batch reader starting from a specific index.